src/lib.rs

```diff
@@ -18,10 +19,63 @@ mod types {
     pub struct Regconfig;
 }
 
+pub mod configuration {
+    use crate::Regconfig;
+
+    use std::io::Write;
+
+    use diesel::backend::Backend;
+    use diesel::deserialize::{self, FromSql};
+    use diesel::serialize::{self, Output};
+    use diesel::sql_types::Integer;
+    use diesel::types::ToSql;
+
+    pub const CONFIGURATION_SIMPLE: u32 = 3748;
+    pub const CONFIGURATION_DANISH: u32 = 12824;
+    pub const CONFIGURATION_DUTCH: u32 = 12826;
+    pub const CONFIGURATION_ENGLISH: u32 = 12828;
+    pub const CONFIGURATION_FINNISH: u32 = 12830;
+    pub const CONFIGURATION_FRENCH: u32 = 12832;
+    pub const CONFIGURATION_GERMAN: u32 = 12834;
+    pub const CONFIGURATION_HUNGARIAN: u32 = 12836;
+    pub const CONFIGURATION_ITALIAN: u32 = 12838;
+    pub const CONFIGURATION_NORWEGIAN: u32 = 12840;
+    pub const CONFIGURATION_PORTUGUESE: u32 = 12842;
+    pub const CONFIGURATION_ROMANIAN: u32 = 12844;
+    pub const CONFIGURATION_RUSSIAN: u32 = 12846;
+    pub const CONFIGURATION_SPANISH: u32 = 12848;
+    pub const CONFIGURATION_SWEDISH: u32 = 12850;
+    pub const CONFIGURATION_TURKISH: u32 = 12852;
+
+    #[derive(Debug, PartialEq, AsExpression)]
+    #[sql_type = "Regconfig"]
+    pub struct TsConfiguration(pub u32);
```

Comment:
If we define the constants above as `TsConfiguration` instead of `u32` we can declare the inner field private. Additionally I would like to make the constants associated constants of this type instead of having them as free constants in this module. So basically something like this:

```rust
impl TsConfiguration {
    pub const SWEDISH: Self = TsConfiguration(12850);
    // …
}
```